### PR TITLE
Prac 05 feedback

### PR DESCRIPTION
**Pull request by Winminoo4444**
Please review this work @jc972271

### CODE REVIEW

File: prac_05/hex_colours.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
+COLOURS_CODES = {
```
Comment (jc972271) [inline prac_05/hex_colours.py:1]: Using the guidelines for the subject dictionaries are named as key_to_value therefore to meet the guideline COLOURS_CODES should be something like COLOURS_TO_CODES

File: prac_05/kivy_layout.kv [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+        Button:
+            text: "Up"
+            on_press: app.handle_press(1)
+        Button:
+            text: "Down"
+            on_press: app.handle_press(-1)
+    BoxLayout:
+        id: names_box
+    Label:
+        text: "I did this :)"
```
Comment (jc972271) [inline prac_05/kivy_layout.kv:20]: Hey just noticed that you missed part of the question about resizing the label to only take up 10% of the window space

File: prac_05/state_names.py [hunk 3]
```diff
@@ -0,0 +1,23 @@
+"""
+CP1404/CP5632 Practical
+State names in a dictionary
+File needs reformatting
+"""
+
+
```
Comment (jc972271) [inline prac_05/state_names.py:7]: Hey, just to let you know code to name isn't in pep8 formatting, according to pep8 dictionaries should go {"key": "value"} with no space for the key and one space for the value

File: prac_05/wimbledon.py [hunk 4]
```diff
@@ -0,0 +1,34 @@
+FILENAME = "wimbledon.csv"
+def main():
+    """Read file"""
```
Comment (jc972271) [inline prac_05/wimbledon.py:3]: Hello, reading the file could be moved to its own function to make main cleaner

File: prac_05/wimbledon.py [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+    with open(FILENAME, "r", encoding="utf-8-sig") as in_file:
+        data = in_file.readlines()
```
Comment (jc972271) [inline prac_05/wimbledon.py:5]: You could import the CSV module as it does some of the things you are doing manually in a single function

File: prac_05/word_occurrences.py [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+lengths = []
+
+text = input("Text: ")
+words = text.split(" ")
+for text in words:
+    if text in words_to_count.keys():
+        words_to_count[text] += 1
+    else:
+        words_to_count[text] = 1
+listed_words = words_to_count.keys()
```
Comment (jc972271) [inline prac_05/word_occurrences.py:11]: listed words is defined only to be used on the next line you could instead go sorted_words(words_to_count.keys())

File: prac_05/wimbledon.py [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+    champions = get_data(data, 2)
+    countries = get_data(data, 1)
+    country_counts = count_item(countries)
+    champion_counts = count_item(champions)
+    print("Wimbledon Champions:")
```
Comment (jc972271) [inline prac_05/wimbledon.py:10]: Printing the results could be refactored out of main to make main cleaner

**Review by jc972271 — COMMENTED** [review]
Just some small things to improve mostly just meeting the pep8 guidelines or making the code easier to read else this is a pretty good attempt at prac5

**Comment by Winminoo4444** [comment]
Thank you for the review.

**Pull request merged**